### PR TITLE
build: Update API-Extractor

### PR DESCRIPTION
**Pull request by Josmithr**
Update to `7.43.1`

### CODE REVIEW

**Review by tylerbutler — APPROVED** [review]

File: build-tools/packages/version-tools/package.json [hunk 1]
```diff
@@ -24,7 +24,11 @@
 	"bin": {
 		"fluv": "./bin/run.js"
 	},
-	"files": ["/bin", "/lib", "/oclif.manifest.json"],
+	"files": [
```
Comment (alexvy86) [inline build-tools/packages/version-tools/package.json:27]: Mixed formatters?

Comment (Josmithr) [reply build-tools/packages/version-tools/package.json:27]: Hmm, looks like. I'll re-run the formatter.

**Review by alexvy86 — APPROVED** [review]

File: build-tools/packages/version-tools/package.json (hunk 1 above)
Comment (tylerbutler) [reply build-tools/packages/version-tools/package.json:27]: Yeah, need to run `pnpm format` from the build-tools root.

**Comment by Josmithr** [comment]
@tylerbutler A test is failing under `build-tools`. Any idea what's going on there?
https://dev.azure.com/fluidframework/public/_build/results?buildId=256366&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=e51c1947-c90a-5cb4-a0bf-0143e1a3235f

**Comment by tylerbutler** [comment]
> @tylerbutler A test is failing under `build-tools`. Any idea what's going on there? [dev.azure.com/fluidframework/public/_build/results?buildId=256366&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=e51c1947-c90a-5cb4-a0bf-0143e1a3235f](https://dev.azure.com/fluidframework/public/_build/results?buildId=256366&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=e51c1947-c90a-5cb4-a0bf-0143e1a3235f)

Should be fixed by this PR: https://github.com/microsoft/FluidFramework/pull/20700

**Review by seanimam — APPROVED** [review]

**Comment by Josmithr** [comment]
@CraigMacomber any thoughts on how we might be able to resolve the build failures in tree?

https://dev.azure.com/fluidframework/public/_build/results?buildId=256365&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=bfe27d5f-9219-57c0-0d67-1b2946a32921&l=50

**Pull request closed without merging**